gcc/jit/jit-recording.cc

```diff
@@ -5188,6 +5188,21 @@ recording::block::get_loc () const
     return NULL;
 }
 
+/* Set the source-location of a block by using that of the first
+   statement within it, if any.  */
+
+void
+recording::block::set_loc (location * loc)
+{
+  recording::statement *stmt =
+      (m_statements.length ())?
+      stmt = m_statements[0]:
+      NULL;
+
+  if (stmt)
+    return stmt->set_loc (loc);
```

Comment:
I would prefer something like this:
```suggestion
  if (m_statements.length () > 0)
      m_statements[0]->set_loc (loc);
```

Reply:
Resolved by deleting  `recording::block::set_block`. Which applies to all other`set_loc`.